### PR TITLE
fix: return GenericVariable for Get/Create Variable methods

### PR DESCRIPTION
**Pull request by aihazm**
Previously, Create/Get Variable methods were returning `Variable`  type those methods should return a `GenericVariable` instead

_Note: the [GetMeasureWithLabel](https://qlik.dev/apis/json-rpc/qix/doc#%23%2Fentries%2FDoc%2Fentries%2FGetMeasureWithLabel) method was also returning the wrong type `Generic` instead of `GenericMeasure`_

This is a breaking change for :
- `CreateSessionVariable`
- `GetVariable`
- `GetVariableById`
- `CreateVariableEx`
and
- `GetMeasureWithLabel`

### CODE REVIEW

File: examples/basics/lists/variable-list/variable-list.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package main
```
Comment (wennmo) [inline examples/basics/lists/variable-list/variable-list.go:1]: We should add this example to the README as well: https://github.com/qlik-oss/enigma-go/blob/master/examples/basics/lists/README.md

**Review by wennmo — COMMENTED** [review]
Think the changes looks good, and great that you added an example for variables 👍 

Since this is a breaking change we will need to include it in a major release of `enigma-go`.

@DnlLrssn, can you take a look at these changes and raise if you have any concerns? Basically there was a mismatch between the hardcoded return types in `enigma-go` compared to that of the QIX definitions.

**Review by DnlLrssn — APPROVED** [review]
👍 looks good from our perspective

**Review by glooms — APPROVED** [review]
I think these changes look great!

One thing, related to this work but not something that should be solved in this PR, is that it would be nice if we could make the structure more flexible to safeguard against these breaking changes as they could possibly happen in the future as well.

**Review by wennmo — APPROVED** [review]
LGTM 🎉 

Lets create a backlog item for improving how these return-types are handled. Preferably enough information should be included in the Open-RPC spec to determine the return type.

**Comment by aihazm** [comment]
> Lets create a backlog item for improving how these return-types are handled. Preferably enough information should be included in the Open-RPC spec to determine the return type.

@wennmo FTR: https://github.com/qlik-oss/enigma-go/issues/192

**Pull request merged**